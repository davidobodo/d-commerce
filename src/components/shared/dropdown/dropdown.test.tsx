import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Dropdown from './dropdown';


afterEach(cleanup);

const defaultProps = {
    options: [],
    selectedOption: jest.fn()
}

it('dropdown should match snapshot', () => {
    const { container } = render(<Dropdown {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
})

it('dropdown should render initially with a default value of Choose an option', () => {
    const { getByLabelText } = render(<Dropdown {...defaultProps} />);
    const input = getByLabelText('input-field') as HTMLInputElement;
    expect(input.value).toBe('Choose an option')
})