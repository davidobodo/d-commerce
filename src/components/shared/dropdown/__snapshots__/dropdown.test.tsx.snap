// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown should match snapshot 1`] = `
<div
  class="sc-AxjAm emNQGb"
  data-testid="dropdown"
>
  <input
    aria-label="input-field"
    value="Choose an option"
  />
  <ul
    data-testid="options"
  >
    <li
      id="Choose an option"
    >
      Choose an option
    </li>
  </ul>
</div>
`;
